### DISCUSSION TITLE
Implement NMS as a postprocessing method after pose inference

### DISCUSSION THREAD

**Discussion by talmo** [talmolab/sleap#1850] [Ideas]
In some cases, we get a lot of highly overlapping but slightly offset poses, particularly in top-down models when the centroid model finds very nearby centroids. (This can also happen with bottom-up, but it's less likely due to the feature map resolution of the PAFs.)

While you can tune the centroid model to reduce the incidence of nearby peaks, an easy way to improve performance in these cases is to apply non-maximal suppression (NMS) on the pose bounding boxes.

This would run after the centroids are found and can be implemented in the compute graph with native TF functions (it's also not slow to do it on CPU, but probably best to avoid additional computation in the centered instance stage anyway).

**Comment by gitttt-1234** [comment]
Closing this discussion, fixed in `sleap-nn`: https://github.com/talmolab/sleap-nn/pull/420!
